### DISCUSSION TITLE
Cask Libreoffice and libreoffice-language-pack

### DISCUSSION THREAD

**Discussion by knebb** [Homebrew/discussions#4944] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.22
ORIGIN: https://github.com/Homebrew/brew
HEAD: 251a098fbf1c02ca95f53b4806e5b6da8d5e4a80
Last commit: 7 days ago
Core tap JSON: 02 Dec 10:04 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1.1-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  dosbox
  openssl@1.1
  sdl_net
  sdl_sound
```

### Description of issue

Hi,

installing the libreoffice cask works fine. After `brew install libreoffice` I start LibreOffice and close it before installing the language pack with `brew install libreoffice-language-pack`.
After this I have my LibreOffice on German language. Perfect so far.

Unfortunately This does not seem to work for upgrades. If I do `brew upgrade` when an update is available I have my LibreOffice verly likely back in English :(

Once happened I have to remove both (libreoffice and libreoffice-langua-pack) with "-f --zap" and re-install with the above steps. But obviously I loose every user setting at this stage.

Is there an easier way to perform an update and keep language settings?

Thanks!

/KNEBB

### Relevant casks

libreoffice

**Comment by jbouttier** [comment]
I have the same issue (with French instead of German). It would be nice to have a smooth, caveat-free, upgrade process for LibreOffice.

**Comment by p-linnane** [comment]
During the upgrade process the previous version is completely removed. Unless the language settings are stored in a file that can persist between upgrades, there's no way to handle this unfortunately.

**Reply by jbouttier** [reply]
Well I believe I do not have any specific language settings to save. When I follow the procedure indicated by @knebb above ("After `brew install libreoffice` I start LibreOffice and close it before installing the language pack with `brew install libreoffice-language-pack`."), I get an interface in my preferred language without me doing any setting. It seems that LibreOffice somehow infers it from the macOS/locale settings. This is the behavior I'd like `brew upgrade` to have too.

**Reply by knebb** [reply]
I totally agree. 
Libreoffice does something when started the first time. when installing LanguagePack afterwards it applies the language installed through the pack.

If this can not be handled through `brew upgrade` might it be an option to copy the users configuration files so the "--zap" does not remove them and copy them back afterwards?
Anyone having knowledge which files to save?
/KNEBB

**Reply by 9764319764319764** [reply]
gh

**Reply by menkej** [reply]
> Well I believe I do not have any specific language settings to save. When I follow the procedure indicated by @knebb above ("After `brew install libreoffice` I start LibreOffice and close it before installing the language pack with `brew install libreoffice-language-pack`."), I get an interface in my preferred language without me doing any setting. It seems that LibreOffice somehow infers it from the macOS/locale settings. This is the behavior I'd like `brew upgrade` to have too.

This worked for me in the past. Now even uninstalling the language pack, starting libreoffice, reinstalling the language pack, does not bring back the localisation. It is still in English. Like this it is unfortunately not really supporting the upgrade process. :-(

**Reply by mordoukna** [reply]
Same here. As a workaround I just downloaded the language pack from https://de.libreoffice.org/download/download/ and tried to install it. Seems to work with the brew version.

**Comment by stultitiophobia** [comment]
same "problem" here ...
are there any ideas how to handle this ?

**Comment by knebb** [comment]
Hi,

I guess a lot of MAc users with non-english primary language have this issue.

Just had the same problem again.
Update available, updated --> English :(

Big problem is that this is *not only cosmetic* because in Calc the formulas have different names: Instead of "=SUMME()" Calc expects (and suggests "=SUM()". While this is fine for =SUM there are oviously loads of other functions affected. THIS IS REALLY CRAP!

So every time when my LO switches back to English I have to do the initially mentioned steps:
```
brew remove libreoffice-language-pack
brew remove --zap libreoffice
brew install libreoffce
[start LibreOffice through GUI and close it again]
brew install libreoffice-language-pack
```
Finally my LO is back to German. But again, this means EVERY user setting is lost, @p-linnane 

This is really no good and I can not believe there is no solution for this....

Greetings

/KNEBB

**Comment by stultitiophobia** [comment]
same here, but I do not need to zap it, just removing the language-package, starting LO GUI and reinstall language-pack fixes it.

```
brew remove libreoffice-language-pack
[start LibreOffice through GUI and close it again]
brew reinstall libreoffice-language-pack
```

this does the job ...
